GDJS/Runtime/events-tools/commontools.ts

```diff
@@ -469,7 +469,7 @@ namespace gdjs {
         angle: float,
         distance: float
       ): number {
-        return distance * Math.cos(gdjs.toRad(angle));
+        return distance !== 0 ? distance * Math.cos(gdjs.toRad(angle)) : 0;
```

Comment:
I've reverted this because this adds an extra check which will be useless in 99% of cases, so would actually slow down in most cases.